syntax = "proto2";

package rv.protos;

import "rv2/protos/machine_learning.proto";
import "rv2/protos/raster_transformer.proto";
import "rv2/protos/raster_source.proto";
import "rv2/protos/annotation_source.proto";
import "rv2/protos/label_item.proto";

message Predict {
    message Project {
        required RasterSource raster_source = 1;
        required AnnotationSource annotation_source = 2;
    }

    message ObjectDetectionOptions {
        optional float merge_thresh = 1 [default=0.5];
        optional float score_thresh = 2 [default=0.5];
    }

    message Options {
        required string model_uri = 1;
        required int32 chip_size = 3;
        optional bool debug = 4 [default=true];
        oneof ml_options_type {
            ObjectDetectionOptions object_detection_options = 5;
        }
    }

    required MachineLearning machine_learning = 1;
    required RasterTransformer raster_transformer = 2;
    repeated LabelItem label_items = 3;
    repeated Project projects = 4;
    required Options options = 5;
}
